From 96ffddcd18bc6c284ae840ba669f910233d45bd0 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:36:41 -0400
Subject: [ethernet] i40e: Don't receive packets when the napi budget == 0

Message-id: <1409658042-22689-41-git-send-email-sassmann@redhat.com>
Patchwork-id: 90233
O-Subject: [RHEL7.1 PATCH 040/281] i40e: Don't receive packets when the napi budget == 0
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Eric W. Biederman" <ebiederm@xmission.com>

Processing any incoming packets with a with a napi budget of 0
is incorrect driver behavior.

This matters as netpoll will shortly call drivers with a budget of 0
to avoid receive packet processing happening in hard irq context.

Signed-off-by: "Eric W. Biederman" <ebiederm@xmission.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 390f86dfbd3e7c4579aaa88281149e1cbac88a2d)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_txrx.c b/drivers/net/ethernet/intel/i40e/i40e_txrx.c
index daa3b29..88666ad 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_txrx.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_txrx.c
@@ -1302,6 +1302,9 @@ static int i40e_clean_rx_irq(struct i40e_ring *rx_ring, int budget)
  u8 rx_ptype;
  u64 qword;
 
+ if (budget <= 0)
+  return 0;
+
  rx_desc = I40E_RX_DESC(rx_ring, i);
  qword = le64_to_cpu(rx_desc->wb.qword1.status_error_len);
  rx_status = (qword & I40E_RXD_QW1_STATUS_MASK) >>
-- 
1.7.1